tests/TypeProviderHelper.php

```diff
@@ -0,0 +1,241 @@
+		'boolean true',
+	];
+
+	/** Keys of all type entries representing an integer. */
+	const GROUP_INT = [
+		'integer 0',
+		'negative integer',
+		'positive integer',
+	];
+
+	/** Keys of all type entries representing a float. */
+	const GROUP_FLOAT = [
+		'float 0.0',
+		'negative float',
+		'positive float',
+	];
+
+	/** Keys of all type entries representing an integer or float. */
+	const GROUP_INT_FLOAT = [
+		'integer 0',
+		'negative integer',
+		'positive integer',
+		'float 0.0',
+		'negative float',
+		'positive float',
+	];
+
+	/** Keys of all type entries representing a string. */
+	const GROUP_STRING = [
+		'empty string',
+		'numeric string',
+		'textual string',
+		'textual string starting with numbers',
+	];
+
+	/** Keys of all type entries which are stringable. */
+	const GROUP_STRINGABLE = [
+		'empty string',
+		'numeric string',
+		'textual string',
+		'textual string starting with numbers',
+		'Stringable object',
+	];
+
+	/** Keys of all type entries representing an array. */
+	const GROUP_ARRAY = [
+		'empty array',
+		'array with values, no keys',
+		'array with values, string keys',
+	];
+
+	/** Keys of all type entries which are iterable. */
+	const GROUP_ITERABLE = [
+		'empty array',
+		'array with values, no keys',
+		'array with values, string keys',
+		'ArrayIterator object',
+		'Iterator object, no array access',
+	];
+
+	/** Keys of all type entries which have array access. */
+	const GROUP_ARRAY_ACCESSIBLE = [
+		'empty array',
+		'array with values, no keys',
+		'array with values, string keys',
+		'ArrayIterator object',
+		'ArrayAccess object',
+	];
+
+	/** Keys of all type entries representing an object. */
+	const GROUP_OBJECT = [
+		'plain object',
+		'Stringable object',
+		'ArrayIterator object',
+		'ArrayAccess object',
+		'Iterator object, no array access',
+	];
+
+	/** Keys of all type entries representing a resource. */
+	const GROUP_RESOURCE = ['resource (dir handle)'];
+
+	/** Keys of all type entries which are considered empty. */
+	const GROUP_EMPTY = [
+		'null',
+		'boolean false',
+		'integer 0',
+		'float 0.0',
+		'empty string',
+		'empty array',
+	];
+
+	/**
+	 * Directory handle.
+	 *
+	 * @var resource
+	 */
+	private static $dir_handle;
+
+	/**
+	 * Clean up after the tests.
+	 *
+	 * This method should be called in the `tear_down_after_class()` of any test class
+	 * using these helper functions.
+	 *
+	 * @return void
+	 */
+	public static function cleanUp() {
+		if (isset(self::$dir_handle)) {
+			closedir(self::$dir_handle);
+		}
+	}
+
+	/**
+	 * Retrieve an array in data provider format with a selection of all typical PHP data types
+	 * *except* the named types specified in the $except parameter.
+	 *
+	 * @param array ...$except One or more arrays containing the names of the types to exclude.
+	 *                         Typically, one or more of the predefined "groups" (see the constants)
+	 *                         would be used here.
+	 *
+	 * @return array
+	 */
```

Comment:
Should we be precise about the structure of the arrays everywhere?
```suggestion
	/**
	 * Retrieve an array in data provider format with a selection of all typical PHP data types
	 * *except* the named types specified in the $except parameter.
	 *
	 * @param array<string> ...$except One or more arrays containing the names of the types to exclude.
	 *                                 Typically, one or more of the predefined "groups" (see the constants)
	 *                                 would be used here.
	 *
	 * @return array<string>
	 */
```